plugins/captcha/powcaptcha/src/Extension/POWCaptcha.php

```diff
@@ -0,0 +1,256 @@
+     *
+     * @return  bool  True if the answer is correct, false otherwise
+     *
+     * @throws  \RuntimeException
+     * @since __DEPLOY_VERSION__
+     */
+    public function onCheckAnswer($code = null)
+    {
+        // Before we verify the actual solution, let's first our challenge key
+        $decoded = base64_decode($code, true);
+
+        // Check for base64 decode errors
+        if (!$decoded) {
+            return false;
+        }
+
+        // Check for json Errors
+        try {
+            $data = json_decode($decoded, true, 2, \JSON_THROW_ON_ERROR);
+        } catch (\JsonException|\ValueError) {
+            return null;
+        }
+
+        // Check for data errors
+        if (!\is_array($data) || empty($data)) {
+            return null;
+        }
+
+        // Invalid salt format
+        if (empty($data['salt']) || !str_contains($data['salt'], 'challengeKey=')) {
+            return null;
+        }
+
+        // Extract challengeKey
+        parse_str(explode("?", $data['salt'])[1], $challengeParams);
+
+        // Check if challengeKey is valid
+        $session = $this->getApplication()->getSession();
+
+        if (!$session->get('plg_captcha_powcaptcha.' . $challengeParams['challengeKey'], false)) {
+            // Key is invalid, return
+            return false;
+        }
+
+        // Key is valid, check for solution
+        if (!$this->getAltcha()->verifySolution((string) $code)) {
+            return false;
+        }
+
+        // Solution was valid, invalidate key
+        $session->set('plg_captcha_powcaptcha.' . $challengeParams['challengeKey'], false);
+
+        // It's valid!
+        return true;
+    }
+
+    /**
+     * Handles the ajax request triggered by altcha to fetch the challenge code
+     *
+     * @param AjaxEvent $event
+     */
+    public function handleAjaxRequest(AjaxEvent $event)
+    {
+        // Altcha expects its challenge code in a specific syntax that is not compatible with com_ajax, raw output
+        @ob_end_clean();
+        header('Content-Type: application/json');
+
+        // CRSF Token check
+        if (!Session::checkToken('get')) {
+            echo json_encode([]);
+            $this->getApplication()->close();
+        }
+
+        // Determine the max number - to be updated in future releases
+        $maxNumber = match ($this->params->get('difficulty', 'moderate')) {
+            "easy"     => self::MAXNUMBER_EASY,
+            "moderate" => self::MAXNUMBER_MODERATE,
+            "hard"     => self::MAXNUMBER_HARD,
+            "custom"   => $this->params->get('maxnumber', 250000)
+        };
+
+        // Calculate expiration time
+        $expiration = Date::getInstance()->add(new \DateInterval('PT' . $this->params->get('expiration', 300) . 'S'));
+
+        // Generate a random key for the challenge - that key is stored in the session and will be checked an invalidated
```

Comment:
something not right here. guessing


```suggestion
        // Generate a random key for the challenge - that key is stored in the session and will be checked and validated
```

Reply:
Updated!